source/includes/monitoring/sdam.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+require __DIR__ . '/vendor/autoload.php';
+
+// start-mysubscriber
+class MySubscriber implements MongoDB\Driver\Monitoring\SDAMSubscriber
+{
+    private $stream;
+    public function __construct($stream)
+    {
+        $this->stream = $stream;
+    }
+    public function serverOpening(MongoDB\Driver\Monitoring\ServerOpeningEvent $event): void
+    {
+        fwrite($this->stream, sprintf(
+            'Server opening on %s:%s%s',
+            $event->getHost(),
+            $event->getPort(),
+            PHP_EOL,
+        ));
+    }
+    public function serverClosed(MongoDB\Driver\Monitoring\ServerClosedEvent $event): void {}
```

Comment:
```suggestion

    public function serverClosed(MongoDB\Driver\Monitoring\ServerClosedEvent $event): void {}
```

Since the methods below have no bodies, I'm fine grouping them together.